src/AasxServerStandardBib/Services-DB/AssetAdministrationShellServiceDB.cs

```diff
@@ -0,0 +1,561 @@
+﻿
+using AasxServer;
+using AasxServerStandardBib.Exceptions;
+using AasxServerStandardBib.Interfaces;
+using AasxServerStandardBib.Logging;
+using AdminShellNS.Extensions;
+using Extensions;
+using Microsoft.IdentityModel.Tokens;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace AasxServerStandardBib.Services
+{
+    public class AssetAdministrationShellServiceDB : IAssetAdministrationShellService
+    {
+        private readonly IAppLogger<AssetAdministrationShellService> _logger;
+        private readonly IAdminShellPackageEnvironmentService _packageEnvService;
+        private readonly IMetamodelVerificationService _verificationService;
+        private readonly ISubmodelService _submodelService;
+        private IDatabase _database;
+
+
+        public AssetAdministrationShellServiceDB(IAppLogger<AssetAdministrationShellService> logger, IAdminShellPackageEnvironmentService packageEnvService, IMetamodelVerificationService verificationService, ISubmodelService submodelService)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger)); ;
+            _packageEnvService = packageEnvService;
+            _verificationService = verificationService;
+            _submodelService = submodelService;
+            _database = Program.database;
+        }
+
+        public IAssetAdministrationShell CreateAssetAdministrationShell(IAssetAdministrationShell body)
+        {
+            //Verify the body first
+            _verificationService.VerifyRequestBody(body);
+
+            var found = _packageEnvService.IsAssetAdministrationShellPresent(body.Id);
+            if (found)
+            {
+                _logger.LogDebug($"Cannot create requested AAS !!");
+                throw new DuplicateException($"AssetAdministrationShell with id {body.Id} already exists.");
+            }
+
+            var output = _packageEnvService.CreateAssetAdministrationShell(body);
+
+            return output;
+        }
+
+        public IReference CreateSubmodelReferenceInAAS(IReference body, string aasIdentifier)
+        {
+            IReference output = null;
+            //Verify request body
+            _verificationService.VerifyRequestBody(body);
+
+            // TODO (jtikekar, 2023-09-04): to check if submodel with requested submodelReference exists in the server
+            var aas = _packageEnvService.GetAssetAdministrationShellById(aasIdentifier, out _);
+
+            if (aas != null)
+            {
+                if (aas.Submodels.IsNullOrEmpty())
+                {
+                    aas.Submodels = new List<IReference>
+                    {
+                        body
+                    };
+                    output = aas.Submodels.Last();
```

Comment:
Consider initializing `aas.Submodels` directly with `body` if it's empty instead of assigning a new list. This approach clarifies the intent and reduces unnecessary reassignments, which can potentially confuse readers. For example:

```csharp
if (aas.Submodels.IsNullOrEmpty())
{
    aas.Submodels = new List<IReference> { body };
    output = body;
}
```